Fix detecting Windows Server 2022 in wxGetOsDescription()

PBfordev opened the pull request:
Windows Server 2022 has build number lower than first Windows 11 build number (20348 vs 22000), so we cannot use Windows 11 build number to test for it here.

Sorry for not noticing this pre-existing issue when working on #24598.

PBfordev committed:
Fix detecting Windows Server 2022 in wxGetOsDescription()

Windows Server 2022 has build number lower than first Windows 11 build
number (20348 vs 22000), so we cannot use Windows 11 build number to
test for it here.

vadz commented on src/msw/utils.cpp:
```diff
@@ -1177,21 +1179,27 @@ wxString wxGetOsDescription()
+                            case WINDOWS_SERVER2022_BUILD:
+                                str = "Windows Server 2022";
+                                break;
+                        }
```
Should we do something like "Windows Server build %d" for the other, unknown, builds? I understand that this is not supposed to happen, but what if they release such build in the future?

vadz reviewed (commented):
Thanks for fixing this, but I think we should use "Windows Server" for the unknown server versions too, either by adding `default:` label as mentioned in the comment or maybe by testing for `wxIsWindowsServer()` in the fallback code below.

PBfordev committed:
Handle better unknown Windows Server versions in wxGetOsDescription()

If unknown Windows version is encountered and it is a server edition,
report it as such.

PBfordev commented:
I opted for adding the information for unknown server editions to the generic fallback code, instead of possibly reporting it as the last known v10.0 server edition.

Unlike workstation editions, server editions seem to have the build number locked, so they are easy to identify. Windows Server 2025 should be released this year and I am guessing that its version will be 10.0 as well, but its build version is not locked yet.

I also considered removing code for dealing with pre-Win7 versions (< 6.1) but I decided to let it stay: Not only It would make backporting harder but I am not sure if wxWidgets 3.3 really cannot run on those systems.
